controllers/networkfence_controller.go

```diff
@@ -18,19 +18,52 @@ package controllers
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+	conn "github.com/csi-addons/kubernetes-csi-addons/internal/connection"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/proto"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/util"
+	"github.com/csi-addons/spec/lib/go/identity"
 )
 
-// NetworkFenceReconciler reconciles a NetworkFence object
+// NetworkFenceReconciler reconciles a NetworkFence object.
 type NetworkFenceReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	Scheme   *runtime.Scheme
+	Connpool *conn.ConnectionPool
+}
+
+const (
+	networkFenceFinalizer = "csiaddons.openshift.io/network-fence"
+)
+
+// validateNetworkFenceSpec validates the NetworkFence spec and checks if values are neither nil nor empty.
+func validateNetworkFenceSpec(nwFence *csiaddonsv1alpha1.NetworkFence) error {
+	if nwFence.Spec.Driver == "" {
+		return errors.New("required parameter driver is not specified")
+	}
+	if nwFence.Spec.Secret.Name == "" {
+		return errors.New("required parameter secret.name is not specified")
+	}
+	if nwFence.Spec.Secret.Namespace == "" {
+		return errors.New("required parameter secret.namespace is not specified")
+	}
```

Comment:
https://github.com/csi-addons/kubernetes-csi-addons/pull/10#discussion_r773646592
optional or mandatory ??

Reply:
Nice catch! This should be caught at ceph-csi layer. Updated